src/report/html.rs

```diff
@@ -152,8 +167,10 @@ fn write_report<W: ReportWriter>(
     dest: &W,
 ) -> Fallible<()> {
     let mut comparison_colors = HashMap::new();
-    let mut result_colors = HashMap::new();
-    let mut result_names = HashMap::new();
+    let mut test_results_to_int = HashMap::new();
+    let mut result_colors = Vec::new();
+    let mut result_names = Vec::new();
+    let mut index: i32 = -1;
```

Comment:
Is there a reason why you didn't use the length of the vector here?

Reply:
Nope, I initially planned to use the length of the hashmap but I couldn't because of borrowing issues so I switched back to a variable without noticing the vector is the same length.